vllm/worker/worker_base.py

```diff
@@ -385,3 +400,20 @@ def execute_method(self, method, *args, **kwargs):
                    "This might cause deadlock in distributed execution.")
             logger.exception(msg)
             raise e
+
+
+def extract_previous_hidden_states(
+        data: Union[ExecuteModelRequest, dict]) -> dict:
+    output = {}
+
+    if isinstance(data, dict):
+        if "previous_hidden_states" in data:
+            output["previous_hidden_states"] = data["previous_hidden_states"]
+    else:
+        if isinstance(data.previous_hidden_states, torch.Tensor):
+            output["previous_hidden_states"] = data.previous_hidden_states
+        elif isinstance(data.previous_hidden_states, HiddenStates):
+            output["previous_hidden_states"] = data.previous_hidden_states\
+                .hidden_states
```

Comment:
Can you explain in a comment why we have such complexity?

Reply:
Will add a comment if needed after resolving https://github.com/vllm-project/vllm/pull/6830/files#r1714091225

Reply:
yeah would be great if this could go away. the less things to unit test the better.